File: order-status-producer/src/main/java/com/training/order/producer/entity/Order.java
package com.training.order.producer.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.ZonedDateTime;
import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "orders")
public class Order {

    @Id
    private UUID id;
    private UUID sagaId;
    private UUID customerId;
    private BigDecimal price;
    @Enumerated(EnumType.STRING)
    private OrderStatus orderStatus;
    private ZonedDateTime createdAt;
    private ZonedDateTime updatedAt;
}

File: order-fe/src/types/order.ts
export type Order = {
  customerId: string
  price: number
}

export type OrderStatus =
  | 'PENDING'
  | 'PAID'
  | 'APPROVED'
  | 'CANCELLED'
  | 'CANCELLING'

export type OrderResponse = {
  id: string
  sagaId: string
  customerId: string
  price: number
  orderStatus: OrderStatus
  createdAt: string
  updatedAt: string
}
